internal/slicer/slicer.go

```diff
@@ -72,33 +83,33 @@ func Run(options *RunOptions) (*Report, error) {
 		targetDirAbs = filepath.Join(dir, targetDir)
 	}
 
+	packages, err := fetchPackages(options)
+	if err != nil {
+		return nil, err
+	}
+	for _, reader := range packages {
+		defer reader.Close()
+	}
+
 	// Build information to process the selection.
+	extract := make(map[string]map[string][]deb.ExtractInfo)
 	for _, slice := range options.Selection.Slices {
 		extractPackage := extract[slice.Package]
 		if extractPackage == nil {
-			archiveName := release.Packages[slice.Package].Archive
-			archive := options.Archives[archiveName]
-			if archive == nil {
-				return nil, fmt.Errorf("archive %q not defined", archiveName)
-			}
-			if !archive.Exists(slice.Package) {
-				return nil, fmt.Errorf("slice package %q missing from archive", slice.Package)
-			}
-			archives[slice.Package] = archive
 			extractPackage = make(map[string][]deb.ExtractInfo)
 			extract[slice.Package] = extractPackage
 		}
-		arch := archives[slice.Package].Options().Arch
+		archiveName := options.Selection.Release.Packages[slice.Package].Archive
```

Comment:
[Comment for reviewer]: The archive map from archive name to archive was redundant and was only used to get the architecture. When we add support for FIPS we will return the architecture as part of the package data but for now I prefer to keep it like this.